feat: Publish ProcessCompleted integration event

BjarkeMeier opened the pull request:
<!--- 🙏 Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Energinet-DataHub/geh-aggregations) before we can accept your contribution. --->

## Description

<!--- Please leave a helpful description of the pull request here. --->

## References

<!--- Are there any issues, pull requests or similar that should be linked here? --->

* #690

Sondergaard commented on source/dotnet/Services/Infrastructure/Integration/ProcessCompleted.proto:
```diff
@@ -0,0 +1,55 @@
+   */
+  string grid_area_code = 2;
+
+  /*
+   * Process types.
+   */
+  enum ProcessType {
+    PT_BALANCEFIXING = 0; // A balance fixing process
+  }
+
+  /*
+   * The process type.
+   */
+  ProcessType process_type = 3;
```
I'm a bit concerned about this enum. If the enum get a new value I would consider it a breaking change for recipients. 

When a recipient took a dependency on this contract they would only expect to receive the value ´PT_BALANCEFIXING´. If they receive an unknown value what should they do?

If you are going to keep the enum, then you need to consider how you would maintain multiple versions later.

BjarkeMeier replied:
I guess `ProcessCompletedV2` would work fine. Are you thinking about other problems or solutions?

Sondergaard replied:
I have two solutions in mind.

1) Being very explicit in the protobuf and have a contract foreach process-type. This approach brings some complexity for handling many similar contracts.

2) Have one single contract that contains the needed fields (batch_id, grid_area_code) and use it as a data structure. When the event is sent on the service bus, the subject field is assigned a value that reflects the event e.g.: BalanceFixingCompleted.

BjarkeMeier replied:
Thx for your input. For now, I believe (1) would incur more work and maintenance without any known value.
I like (2). I'll discuss it with the team and possibly implement it in an upcoming PR.

BjarkeMeier replied:
@lasrinnil , @JesperGomes what are your thoughts?

lasrinnil replied:
I have no issue with this approach as I see it, it leans into the KISS principle.

I personally would prefer to begin with a simple contract, get feedback and act on it accordingly.

Who even knows if we have all the needed bits and bobs in the event, to begin with :)

JesperGomes replied:
I think what is in this PR is not too bad. The contract is very transparent. I guess the contract is less transparent if we rely of a specific subejct field content? Also, I guess it won't remove the problem completely: the recipient might still want to consider what to do when the subject field changes.

JesperGomes replied:
...but from a semantic versioning point of view I see the point. I would be okay with either.

BjarkeMeier replied:
Thx for your input. I've added a task to consider switching to the suggested solution (2).

JesperGomes reviewed (dismissed):
Looks good to me. Only minor comments. I see that Codecov is warning about coverage at several places. Are we missing some (integration) test?

BjarkeMeier commented on source/dotnet/Services/ProcessManager/Endpoints/PublishProcessCompletedIntegrationEventEndpoint.cs:
```diff
@@ -0,0 +1,54 @@
+        await _processApplicationService.PublishProcessCompletedIntegrationEventsAsync(processCompletedEvent).ConfigureAwait(false);
+    }
+
+    private async Task<T> DeserializeByteArrayAsync<T>(byte[] data)
```
Agree. I'll address it in another PR.

BjarkeMeier commented:
> Looks good to me. Only minor comments. I see that Codecov is warning about coverage at several places. Are we missing some (integration) test?

Thx @JesperGomes. I've added tests to resolve all annotations. I think I've added the one integration test that I'd like to add. Would you like to see others? Can you approve the PR as-is?
